asserts/device_asserts.go

```diff
@@ -218,6 +222,29 @@ var (
 	classicModelOptional = []string{"architecture", "gadget"}
 )
 
+var almostValidName = regexp.MustCompile("^[a-z0-9-]*[a-z][a-z0-9-]*$")
+
+// validateSnapName checks whether the name can be used as a snap name
+//
+// This function should be synchronized with the reference implementation
+// snap.ValidateName() in snap/validate.go
+func validateSnapName(name string) error {
+	isValidName := func() bool {
+		if !almostValidName.MatchString(name) {
+			return false
+		}
+		if name[0] == '-' || name[len(name)-1] == '-' || strings.Contains(name, "--") {
+			return false
+		}
+		return true
+	}
+
+	if len(name) > 40 || !isValidName() {
+		return fmt.Errorf("invalid snap name: %q", name)
```

Comment:
would be good to follow the style of errors here and have  "invalid snap name in header %q: %s" in this